src/js/tracks/text-track-settings.js

```diff
@@ -273,25 +242,184 @@ class TextTrackSettings extends Component {
+  createElSelect_(key) {
+    const config = selectConfigs[key];
+    const id = config.id.replace('%s', this.id_);
+
+    return [
+      createEl(
+        'label',
+        {className: 'vjs-label', textContent: config.label},
+        {for: id}),
+      createEl(
+        'select',
+        {id},
+        undefined,
+        config.options.map(o => {
+          return createEl('option', {textContent: o[1], value: o[0]});
+        }))
+    ];
+  }
+
+  /**
+   * Create color elements for the component
+   *
+   * @private
+   * @return {Element}
+   * @method createElColors_
+   */
+  createElColors_() {
+    return createEl(
+      'div',
+      {className: 'vjs-tracksettings-colors'},
+      undefined,
+      [
+        createEl(
+          'fieldset',
+          {className: 'vjs-fg-color vjs-tracksetting'},
+          undefined,
+          [createEl('legend', {textContent: 'Text'})].concat(
+            this.createElSelect_('color'),
+            createEl(
+              'span',
+              {className: 'vjs-text-opacity vjs-opacity'},
+              undefined,
+              this.createElSelect_('textOpacity')))),
+        createEl(
+          'fieldset',
+          {className: 'vjs-bg-color vjs-tracksetting'},
+          undefined,
+          [createEl('legend', {textContent: 'Background'})].concat(
+            this.createElSelect_('backgroundColor'),
+            createEl(
+              'span',
+              {className: 'vjs-bg-opacity vjs-opacity'},
+              undefined,
+              this.createElSelect_('backgroundOpacity')))),
+        createEl(
+          'fieldset',
+          {className: 'vjs-window-color vjs-tracksetting'},
+          undefined,
+          [createEl('legend', {textContent: 'Window'})].concat(
+            this.createElSelect_('windowColor'),
+            createEl(
+              'span',
+              {className: 'vjs-window-opacity vjs-opacity'},
+              undefined,
+              this.createElSelect_('windowOpacity'))))
+      ]);
+  }
+
+  /**
+   * Create font elements for the component
+   *
+   * @private
+   * @return {Element}
+   * @method createElFont_
+   */
+  createElFont_() {
+    return createEl(
+      'div',
+      {className: 'vjs-tracksettings-font'},
+      undefined,
+      [
+        createEl(
+          'div',
+          {className: 'vjs-font-percent vjs-tracksetting'},
+          undefined,
+          this.createElSelect_('fontPercent')),
+        createEl(
+          'div',
+          {className: 'vjs-edge-style vjs-tracksetting'},
+          undefined,
+          this.createElSelect_('edgeStyle')),
+        createEl(
+          'div',
+          {className: 'vjs-font-family vjs-tracksetting'},
+          undefined,
+          this.createElSelect_('fontFamily'))
+      ]);
+  }
+
+  /**
+   * Create controls for the component
+   *
+   * @private
+   * @return {Element}
+   * @method createElControls_
+   */
+  createElControls_() {
+    return createEl(
+      'div',
```

Comment:
same comment with this one as the createElColors_ although it isn't nearly as bad.